### DISCUSSION TITLE
Inconsistency in Values of AuxVariables after Transfer using MultiAppCopyTransfer

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#27043] [Q&A General] [answered]
Dear moose experts,

I am writing to address an issue with numerical inconsistency during the transfer of an AuxVariable named `df_elastic_dgr0` from a sub-cp case to a main-pf case within our multi-app case.

Specifically, I have initiated a multi-app scenario, comprising [02_parent_pf.i and 02_sub_cp.i](https://github.com/PengWei97/yinglong/commit/6217ce7cecb11ed7da7b55a6ceaa307a91efbb9e#diff-437f8a8c4cfd90e54dee80a49df75f11390a56b8af557cf49807ca3c2b9029c9). The objective is to facilitate the transfer of the MaterialProperty df_elastic/dgr0 from the sub-cp simulation to the main-pf simulation. To execute this, I initially utilized the auxkernel MaterialRealAux within the sub-cp simulation to generate the auxVariable df_elastic_dgr0. Subsequently, employing the MultiAppCopyTransfer, I transferred df_elastic_dgr0 to the main-pf simulation. Despite sharing the identical name, df_elastic_dgr0, it has come to my attention that there exists a discrepancy between the values of df_elastic_dgr0 observed in the sub-cp and main-pf simulations, as illustrated in the attached figure.

![image](https://github.com/idaholab/moose/assets/56183775/20b8e198-db08-48f4-b133-66f510a7b354)

This discrepancy is concerning and may indicate an issue either in the configuration of the MultiAppCopyTransfer or in the handling of the AuxVariable within the simulations. I would greatly appreciate any insights, advice, or examples you can provide to help me address this challenge effectively. Your expertise in this matter is invaluable, and your guidance will significantly impact the success of my project.

Thank you in advance for your time and support. I look forward to your response and guidance.

Best regards,

Wei

**Comment by GiudGiud** [comment, marked as answer]
Hello

If the meshes are identical and the variables also of the same finite element type, then you can try other transfers. 

MultiAppGeneralFieldNearesLocation and ShapeEvaluation transfer should both give you the same result (if the variable is constant monomial or a first order Lagrange for NearestLocation, any variable for ShapeEvaluation).

When is the transfer executed? And when is the output performed? Which time step are we looking at?
Are you using block restriction or any other more advanced parameters to the transfer?

Guillaume

**Reply by PengWei97** [reply]
Dear Guillaume,

First and foremost, I would like to express my gratitude for your suggestions and assistance, Following your advice, we successfully resolved the issue we were facing by using MultiAppGeneralFieldNeastLocationTransfer. Additionally, we observed that the use of the interval parameter in the outputs module significantly affects the comparison of output results.

Thank you once again for your guidance, which has been crucial to the progress of our projects.

Best wishes!!

wei
